src/main/webapp/app/shared/metis/postings-create-edit-modal/post-create-edit-modal/post-create-edit-modal.component.ts

```diff
@@ -79,4 +116,43 @@ export class PostCreateEditModalComponent extends PostingsCreateEditModalDirecti
             this.modalTitle = 'artemisApp.metis.createModalTitlePost';
         }
     }
+
+    /**
+     * required for distinguishing different select options for the context selector,
+     * Angular needs to be able to identify the currently selected option
+     */
+    compareContextSelectorOptionFn(option1: ContextSelectorOption, option2: ContextSelectorOption): boolean {
```

Comment:
since the same function is defined for selector. Is there anyway to defined this method only once and using it in several places in more efficient way? I'm just asking to know what are the options we have, since you are experienced with Angular

Reply:
Thanks for taking the time and checking the code, I really appreciate it!
Regarding the comment, it seems to the same and therefore can be handled with one function. I thought about it and tried to figure it out. The `compareContextSelectorOption` is the helper function for selecting contexts when creating a post (therefore it's in `post-create-edit-modal.component.ts`) -> a user can select one of the course-wide topics, or a specific lecture/exercise.
The `compareContextFilterOption` is the helper function for choosing the context a user wants to filter the course posts in the overview for (therefore it's in `course-discussion.component.ts`) -> a user can select one of the course-topics, a specific lecture/exercise _OR the course_. The latter is the different compared to the `compareContextSelectorOption` as the course itself is not an option to chose from when creating a post. 
To put it in a nutshell, the interfaced types that are compared in these two - similar looking - functions differ:
```
interface ContextFilterOption {
    courseId?: number;
    lectureId?: number;
    exerciseId?: number;
    courseWideContext?: CourseWideContext;
}

vs.

interface ContextSelectorOption {
    lecture?: Lecture;
    exercise?: Exercise;
    courseWideContext?: CourseWideContext;
}
````
I think is ok to leave it as it is for now. Maybe I could try to merge the interfaces, but I am not so sure about it yet. Tough, I'll keep it in mind for the following PRs.